projects/RabbitMQ.Client/client/impl/Session.cs

```diff
@@ -43,24 +43,24 @@
 namespace RabbitMQ.Client.Impl
 {
     ///<summary>Normal ISession implementation used during normal channel operation.</summary>
-    class Session : SessionBase
+    internal class Session : SessionBase
     {
         private readonly CommandAssembler _assembler;
 
-        public Session(Connection connection, int channelNumber)
+        public Session(Connection connection, ushort channelNumber)
             : base(connection, channelNumber)
         {
             _assembler = new CommandAssembler(connection.Protocol);
         }
 
         public override void HandleFrame(in InboundFrame frame)
         {
-            Command cmd = _assembler.HandleFrame(in frame);
-            if (cmd != null)
+            var cmd = _assembler.HandleFrame(in frame);
+            if (!cmd.IsEmpty)
             {
                 using (cmd)
                 {
-                    OnCommandReceived(cmd);
+                    CommandReceived.Invoke(in cmd);
```

Comment:
Why not

```
CommandReceived(in cmd)
```

?

Reply:
I would have prefered

```
            if (cmd.IsEmpty)
            {
                return;
            }

            using (cmd)
            {
                CommandReceived.Invoke(in cmd);
            }
```

Reply:
Couldn't this technically be `null`? Should we always assign a default delegate so that we never need to null check?

Reply:
Just my style, I don't mind adapting to whatever style there is preferred here.

> Couldn't this technically be `null`? Should we always assign a default delegate so that we never need to null check?

technically yes, as it's not set in the constructor, but it's done in Session.Initialise which is called from the constructor. In that sense no it can not.